pkg/common/cns-lib/volume/manager.go

```diff
@@ -615,6 +615,17 @@ func (m *defaultManager) QueryVolume(ctx context.Context, queryFilter cnstypes.C
 		log.Errorf("CNS QueryVolume failed from vCenter %q with err: %v", m.virtualCenter.Config.Host, err)
 		return nil, err
 	}
+	if m.virtualCenter.Client.Version == cns.ReleaseVSAN67u3 {
+		for volumeIndex, volume := range res.Volumes {
```

Comment:
One nit: How about just putting these lines in a function and calling that from Query and QueryAll?

Reply:
done - https://github.com/kubernetes-sigs/vsphere-csi-driver/pull/409/commits/731db74276fb0b88fca9ffb213f9d849d7343e04